Named ValueTuples throw exception

When defining a shape like this:

```csharp
[GenerateShape]
public sealed partial class Message
{
    public (string A, string B) Foo { get; set; }
}
```

The following error is produced on `PolyType` version `0.56.1`:

```text
warning CS8785: Generator 'PolyTypeGenerator' failed to generate source. It will not contribute to the output and compilation errors may occur as a result. Exception was of type 'KeyNotFoundException' with message 'The given key '(string A, string B)' was not present in the dictionary.'.
```

It appears it has something to do with the fact that the tuple is named. If I remove the names, it works fine:
```csharp
[GenerateShape]
public sealed partial class Message
{
    public (string, string) Foo { get; set; }
}
```
or:
```csharp
[GenerateShape]
public sealed partial class Message
{
    public KeyValuePair<string, string> Foo { get; set; }
}
```

--- a/src/PolyType.SourceGenerator/Parser/Parser.ModelMapper.cs
+++ b/src/PolyType.SourceGenerator/Parser/Parser.ModelMapper.cs
@@ -338,7 +338,7 @@ public sealed partial class Parser
                 _ => false
             },
 
-            PropertyType = CreateTypeId(property.PropertyType),
+            PropertyType = CreateTypeId(NormalizeType(property.PropertyType)),
             IsGetterNonNullable = emitGetter && property.IsGetterNonNullable,
             IsSetterNonNullable = emitSetter && property.IsSetterNonNullable,
             PropertyTypeContainsNullabilityAnnotations = property.PropertyType.ContainsNullabilityAnnotations(),

--- a/tests/PolyType.SourceGenerator.UnitTests/CompilationTests.cs
+++ b/tests/PolyType.SourceGenerator.UnitTests/CompilationTests.cs
@@ -1414,4 +1414,25 @@ public static partial class CompilationTests
         PolyTypeSourceGeneratorResult result = CompilationHelpers.RunPolyTypeSourceGenerator(compilation);
         Assert.Empty(result.Diagnostics);
     }
+
+    [Fact]
+    public static void NamedTupleProperties()
+    {
+        // Regression test for https://github.com/eiriktsarpalis/PolyType/issues/241
+        Compilation compilation = CompilationHelpers.CreateCompilation("""
+            using PolyType;
+            using System.Collections.Generic;
+
+            [GenerateShape]
+            public sealed partial class Message
+            {
+                public (string A, string B) Foo { get; set; }
+                public List<(string X, int Y)>? Bar { get; set; }
+                public Dictionary<(int I, string S), (string U, string V)>? Baz { get; set; }
+            }
+            """);
+
+        PolyTypeSourceGeneratorResult result = CompilationHelpers.RunPolyTypeSourceGenerator(compilation);
+        Assert.Empty(result.Diagnostics);
+    }
 }

